### PR TITLE
CVSL-1711 Bump dependencies

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ parameters:
     default: the_elephants_devs
   java-version:
     type: string
-    default: "19.0"
+    default: "21.0"
```
Comment (andrewrlee) [inline .circleci/config.yml:16]: move to java 21

File: detekt-baseline.xml [hunk 2]
```diff
@@ -48,14 +48,12 @@
...
     <ID>MagicNumber:ProbationSearchApiClient.kt$ProbationSearchApiClient$2000</ID>
     <ID>MagicNumber:SubjectAccessRequestController.kt$SubjectAccessRequestController$209</ID>
-    <ID>MatchingDeclarationName:AppointmentWithType.kt$AppointmentPersonType</ID>
     <ID>MatchingDeclarationName:EventSpecifications.kt$EventQueryObject</ID>
     <ID>MatchingDeclarationName:LicenceFunctions.kt$SentenceChanges</ID>
     <ID>MatchingDeclarationName:LicenceSpecifications.kt$LicenceQueryObject</ID>
     <ID>ReturnCount:ExclusionZoneService.kt$ExclusionZoneService$fun createThumbnailImageJpeg(fullSizeImage: ByteArray?, width: Int = 150, height: Int = 200): ByteArray?</ID>
     <ID>ReturnCount:StaffService.kt$StaffService$@Transactional fun updateComDetails(comDetails: UpdateComRequest): CommunityOffenderManager</ID>
     <ID>SpreadOperator:CreateAndVaryALicenceApi.kt$(*args)</ID>
-    <ID>SwallowedException:EventService.kt$EventService$e: PropertyReferenceException</ID>
```
Comment (andrewrlee) [inline detekt-baseline.xml:58]: fixed a couple of warnings

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-8.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.6-bin.zip
```
Comment (andrewrlee) [inline gradle/wrapper/gradle-wrapper.properties:3]: bumped gradle version

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/kotlinjpaspecificationdsl/JPASpecificationDSL.kt [hunk 4]
```diff
@@ -35,7 +35,7 @@ fun <T, R> KProperty1<T, R?>.equal(x: R): Specification<T> = spec { equal(it, x)
 fun <T, R> KProperty1<T, R?>.notEqual(x: R): Specification<T> = spec { notEqual(it, x) }
 
 // Ignores empty collection otherwise an empty 'in' predicate will be generated which will never match any results
-fun <T, R : Any> KProperty1<T, R?>.`in`(values: Collection<R>): Specification<T> = if (values.isNotEmpty()) {
+fun <T, R : Any> KProperty1<T, R?>.includedIn(values: Collection<R>): Specification<T> = if (values.isNotEmpty()) {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/kotlinjpaspecificationdsl/JPASpecificationDSL.kt:38]: helping out ktlint

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EventService.kt [hunk 5]
```diff
@@ -17,7 +17,7 @@ class EventService(private val licenceEventRepository: LicenceEventRepository) {
       val matchingEvents = licenceEventRepository.findAll(eventQueryObject.toSpecification(), eventQueryObject.getSort())
       return matchingEvents.transformToModelEvents()
     } catch (e: PropertyReferenceException) {
-      throw ValidationException(e.message)
+      throw ValidationException(e.message, e)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EventService.kt:20]: removing swallowed exception

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt [hunk 6]
```diff
@@ -12,9 +12,9 @@ import java.time.LocalDate
 class ReleaseDateService(
   private val bankHolidayService: BankHolidayService,
   private val clock: Clock,
-  @Value("\${maxNumberOfWorkingDaysAllowedForEarlyRelease:3}") private val maxNumberOfWorkingDaysAllowedForEarlyRelease: Int,
-  @Value("\${maxNumberOfWorkingDaysToTriggerAllocationWarningEmail:6}") private val maxNumberOfWorkingDaysToTriggerAllocationWarningEmail: Int,
-  @Value("\${maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus:3}") private val maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus: Int,
+  @Value("\${maxNumberOfWorkingDaysAllowedForEarlyRelease:3}") private val maxNumberOfWorkingDaysAllowedForEarlyRelease: Int = 3,
+  @Value("\${maxNumberOfWorkingDaysToTriggerAllocationWarningEmail:6}") private val maxNumberOfWorkingDaysToTriggerAllocationWarningEmail: Int = 6,
+  @Value("\${maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus:3}") private val maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus: Int = 3,
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateService.kt:17]: adding defaults matching spEl defaults

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/prison/PrisonApiClient.kt [hunk 7]
```diff
@@ -20,7 +20,7 @@ class PrisonApiClient(@Qualifier("oauthPrisonClient") val prisonerApiWebClient:
   fun getHdcStatus(bookingId: Long): Mono<PrisonerHdcStatus> {
     return prisonerApiWebClient
       .get()
-      .uri("/offender-sentences/booking/$bookingId/home-detention-curfews/latest")
+      .uri("/offender-sentences/booking/{bookingId}/home-detention-curfews/latest", bookingId)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/prison/PrisonApiClient.kt:23]: this is addressing a minor memory leak - each unique url creates an instance of a long running micrometer Tag classes - so one for ever booking id that is passed here! 

But using parameterised routes this avoids that. 
For more details there's a [slack thread](https://mojdt.slack.com/archives/C69NWE339/p1697456808633699)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ControllerTagTest.kt [hunk 8]
```diff
@@ -10,6 +10,12 @@ import java.lang.reflect.AnnotatedElement
 import java.lang.reflect.Method
 
 class ControllerTagTest {
+  private val controllersToIgnore = listOf(
+    // These only gets wired up in tests
+    "uk.gov.justice.hmpps.sqs.HmppsQueueResource",
+    "uk.gov.justice.hmpps.sqs.HmppsReactiveQueueResource",
+  )
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/privateApi/ControllerTagTest.kt:17]: This is a new class in latest SQS lib

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt [hunk 9]
```diff
@@ -342,9 +339,7 @@ class ReleaseDateServiceTest {
       LocalDate.parse("2018-12-03"),
       LocalDate.parse("2018-12-04"),
     )
-    const val maxNumberOfWorkingDaysAllowedForEarlyRelease = 3
-    const val maxNumberOfWorkingDaysToTriggerAllocationWarningEmail = 6
-    const val maxNumberOfWorkingDaysToUpdateLicenceTimeOutStatus = 3
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ReleaseDateServiceTest.kt:347]: letting it default to default args

File: build.gradle.kts [hunk 10]
```diff
@@ -57,14 +57,14 @@ dependencies {
...
+  runtimeOnly("org.postgresql:postgresql:42.7.2")
 
   implementation("com.google.code.gson:gson:2.10.1")
   implementation("io.arrow-kt:arrow-core:1.2.1")
-  implementation("io.hypersistence:hypersistence-utils-hibernate-60:3.5.1")
+  implementation("io.hypersistence:hypersistence-utils-hibernate-60:3.7.3")
 
   // SQS/SNS dependencies
-  implementation("uk.gov.justice.service.hmpps:hmpps-sqs-spring-boot-starter:2.2.1")
+  implementation("uk.gov.justice.service.hmpps:hmpps-sqs-spring-boot-starter:3.1.1")
```
Comment (andrewrlee) [inline build.gradle.kts:67]: moving to java 21 meant we could adopt this.

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Pull request merged**